src/utils.ts

```diff
@@ -0,0 +1,7 @@
+import camelcaseKeys = require('camelcase-keys');
+
+export const camelizeResponse = (response: any) =>
```

Comment:
I'm thinking we should add a few tests for `camelizeResponse()`. It might feel like we are retesting the implementation of `camelcase-keys`, but really the tests should ensure that `camelizeResponse()` meets the requirements of our program regardless of what is implemented in `camelizeResponse()`. Does that make sense?

Reply:
Imagine a scenario, where `deep: true` was changed to `deep: false`. 

If this happens using the current PR, the test failure output doesn't explicitly point to where/why the failure is occurring. You only see failures via side-effects, as seen below.

All tests pass when removing the `exclude` option. Should our build fail if that option isn't present? If not, is that option a necessary?

```
yarn test
yarn run v1.5.1
$ jest
 PASS  src/request/device.spec.ts
 PASS  src/request/transaction.spec.ts
 FAIL  src/webServiceClient.spec.ts
  ● WebServiceClient › score() › handles "full" responses

    expect(received).toEqual(expected)

    Expected value to equal:
      {"disposition": {"action": "accept", "reason": "default"}, "fundsRemaining": 25, "id": "5bc5d6c2-b2c8-40af-87f4-6d61af86b6ae", "ipAddress": {"risk": 0.01}, "queriesRemaining": 5000, "riskScore": 0.01, "warnings": [{"code": "INPUT_INVALID", "input_pointer": "/shipping/city", "warning": "Encountered value at /shipping/city that does not meet the required constraints"}]}
    Received:
      {"disposition": {"action": "accept", "reason": "default"}, "fundsRemaining": 25, "id": "5bc5d6c2-b2c8-40af-87f4-6d61af86b6ae", "ipAddress": {"risk": 0.01}, "queriesRemaining": 5000, "riskScore": 0.01, "warnings": [{"code": "INPUT_INVALID", "inputPointer": "/shipping/city", "warning": "Encountered value at /shipping/city that does not meet the required constraints"}]}

    Difference:

    - Expected
    + Received

    @@ -1,6 +1,6 @@
    - Object {
    + Score {
        "disposition": Object {
          "action": "accept",
          "reason": "default",
        },
        "fundsRemaining": 25,
    @@ -11,10 +11,10 @@
        "queriesRemaining": 5000,
        "riskScore": 0.01,
        "warnings": Array [
          Object {
            "code": "INPUT_INVALID",
    -       "input_pointer": "/shipping/city",
    +       "inputPointer": "/shipping/city",
            "warning": "Encountered value at /shipping/city that does not meet the required constraints",
          },
        ],
      }

      30 |         .reply(200, fixtures.response.full);
      31 |
    > 32 |       return expect(client.score(transaction)).resolves.toEqual(
         |                                                         ^
      33 |         camelizeResponse(fixtures.response.full)
      34 |       );
      35 |     });

      at Object.toEqual (node_modules/expect/build/index.js:172:22)
      at Object.it (src/webServiceClient.spec.ts:32:57)

  ● WebServiceClient › score() › handles "no disposition" responses

    expect(received).toEqual(expected)

    Expected value to equal:
      {"fundsRemaining": 25, "id": "5bc5d6c2-b2c8-40af-87f4-6d61af86b6ae", "ipAddress": {"risk": 0.01}, "queriesRemaining": 5000, "riskScore": 0.01, "warnings": [{"code": "INPUT_INVALID", "input_pointer": "/shipping/city", "warning": "Encountered value at /shipping/city that does not meet the required constraints"}]}
    Received:
      {"disposition": undefined, "fundsRemaining": 25, "id": "5bc5d6c2-b2c8-40af-87f4-6d61af86b6ae", "ipAddress": {"risk": 0.01}, "queriesRemaining": 5000, "riskScore": 0.01, "warnings": [{"code": "INPUT_INVALID", "inputPointer": "/shipping/city", "warning": "Encountered value at /shipping/city that does not meet the required constraints"}]}

    Difference:

    - Expected
    + Received

    - Object {
    + Score {
    +   "disposition": undefined,
        "fundsRemaining": 25,
        "id": "5bc5d6c2-b2c8-40af-87f4-6d61af86b6ae",
        "ipAddress": Object {
          "risk": 0.01,
        },
        "queriesRemaining": 5000,
        "riskScore": 0.01,
        "warnings": Array [
          Object {
            "code": "INPUT_INVALID",
    -       "input_pointer": "/shipping/city",
    +       "inputPointer": "/shipping/city",
            "warning": "Encountered value at /shipping/city that does not meet the required constraints",
          },
        ],
      }

      55 |         .reply(200, fixtures.response.noDisposition);
      56 |
    > 57 |       return expect(client.score(transaction)).resolves.toEqual(
         |                                                         ^
      58 |         camelizeResponse(fixtures.response.noDisposition)
      59 |       );
      60 |     });

      at Object.toEqual (node_modules/expect/build/index.js:172:22)
      at Object.it (src/webServiceClient.spec.ts:57:57)
```

Reply:
> Imagine a scenario, where deep: true was changed to deep: false.
>
> If this happens using the current PR, the test failure output doesn't explicitly point to where/why the failure is occurring. You only see failures via side-effects, as seen below.

It's a similar scenario with private functions, and we don't write unit tests for those.

> All tests pass when removing the exclude option. Should our build fail if that option isn't present? If not, is that option a necessary?

It's a carry over from the GeoIP2 web records, which will eventually make its way into this library.  I'm fine with removing it now and adding it later when needed, but I figure there's no harm having it in there.

Reply:
I think I see where you're coming from. With this in mind, will we generally avoid writing any tests for anything in `utils/*`?

Reply:
It's hard to make a general assumption about `utils/*`, soI'd lean on our coverage reports to guide us.